web/src/compose.js

```diff
@@ -706,6 +734,27 @@ export function initialize() {
         }
     });
 
+    $("body").on("click", ".audio_link", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        let $target_textarea;
+        let edit_message_id;
+        if ($(e.target).parents(".message_edit_form").length === 1) {
+            edit_message_id = rows.id($(e.target).parents(".message_row"));
+            $target_textarea = $(`#edit_form_${CSS.escape(edit_message_id)} .message_edit_content`);
+        }
+
+        const show_audio_chat_button = compute_show_audio_chat_button();
+        if (!show_audio_chat_button) {
+            return;
+        }
+
+        const audio_call_link =
+            generate_jitsi_meet_video_call_link() + "#config.startWithVideoMuted=true";
```

Comment:
This logic only works for Jitsi, so it shouldn't be written this way. 

Ideally, rather than having this be so parallel to the video call code path, we'd have a shared function like `generate_jitsi_meet_call_link` that takes a parameter for whether you want a video or audio call ... or ideally, a broader refactor that extracts the click handler for the video call link as a function that accepts a parameter for whether a video or audio -- that will save us a ton of time implementing the audio call feature for Zoom, for example, which likely just involves adding a similar small change to the 100-line function for asking Zoom to make a call.

Reply:
Agreed. That would make it easier to expand this feature to other providers as well.